api/types/authority.go

```diff
@@ -33,6 +33,8 @@ import (
 type CertAuthority interface {
 	// ResourceWithSecrets sets common resource properties
 	ResourceWithSecrets
+	// SetMetadata sets CA metadata
+	SetMetadata(meta Metadata)
```

Comment:
Any reason why `SetMetadata` was included on specific resources and not part of the `Resource` interface (where `GetMetadata` lives)?

Reply:
@russjones I'd have to add this method to every single resource then, wanted to avoid this large refactor in this PR